### PR TITLE
Convert SelectList to Typescript

### PR DESCRIPTION
**Pull request by iethree**
## Description

- converts SelectList component and subcomponents to typescript
- adds unit tests
- adds storybook story

Screen Shot 2023-01-26 at 3 44 45 PM

### CODE REVIEW

File: frontend/src/metabase/hooks/use-scroll-on-mount.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export const useScrollOnMount = () => {
 
   useEffect(() => {
     if (ref.current) {
-      ref.current.scrollIntoView({ block: "center" });
+      ref.current.scrollIntoView?.({ block: "center" });
```
Comment (iethree) [inline frontend/src/metabase/hooks/use-scroll-on-mount.js:8]: this change was necessary to make unit tests pass, I don't think jest-dom knows about scrolling into view, so it kept throwing "this is not a function"

File: frontend/src/metabase/components/SelectList/SelectList.unit.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        <SelectList.Item
+          id="2"
+          name="Item 2"
+          icon="check"
+          onSelect={selectSpy}
+        />
+      </SelectList>,
+    );
+
+    await userEvent.click(screen.getByText("Item 2"));
```
Comment (kulyk) [inline frontend/src/metabase/components/SelectList/SelectList.unit.spec.tsx:56]: AFAIK `userEvent.click` isn't async

File: frontend/src/metabase/components/SelectList/SelectList.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+      expect(screen.getByText("Item 1")).toBeInTheDocument();
+    });
+
+    it("renders the icon of the item", () => {
+      render(
+        <SelectList.Item id="1" name="Item 1" icon="check" onSelect={_.noop} />,
+      );
+
+      expect(screen.getByLabelText("check icon")).toBeInTheDocument();
```
Comment (kulyk) [inline frontend/src/metabase/components/SelectList/SelectList.unit.spec.tsx:75]: nip: you can use `getIcon` / `queryIcon` helpers from `__support__/ui`

**Review by kulyk — APPROVED** [review]
LGTM, just a few minor comments on tests

**Pull request merged**